### DISCUSSION TITLE
Does 'single include dependency compile' exists

### DISCUSSION THREAD

**Discussion by ppyck** [vscode-abl/vscode-abl#239] [Q&A]
Is there a possibility in vscode to tell the ABL Language Server that you want a 'single include dependency compile' ? If so how do I turn this on ?

**Comment by mikefechner** [comment]
You mean compile everything that depends on a certain include file?

**Comment by ppyck** [comment]
@mikefechner  No, just the opposite. Now when I change an include file (*.i) and I save, the Language Server automatically start to compile everything that depends on the include file. 
What I would like is to turn this off, so I can compile the source I want to test it on, do the test and if ok then tell the ABL Language Server to compile everything depending on that include file
(cfr. the 'Single include dependency compile' you can put on in the outline of the oedt-editor in PDSOE)

**Comment by gquerret** [comment]
There's no such thing in the language server. What is the point ? Avoid compiling too many files ?

**Comment by ppyck** [comment]
Indeed, that was the point. I had a problem in a source which was due to an include (that include is used in more than 3000 sources). After saving the include, I wanted to test, but unfortunately the source with the problem was one of the lasts who got compiled. So I did 4 tests against a source still compiled with the old version of the include,  before the new .r was written to disk

**Comment by gquerret** [comment]
Saving .p trigger compilation with higher priority than the 3000 files recompiled due to the include change.

**Comment by ppyck** [comment]
Am I understanding you correct? Although saving the include started the recompile op more than 3000 files, if during that compile I would have saved the source I wanted to test on, that compile would have been done inbetween the ongoing compile because it has a higher priority ?

**Reply by gquerret** [reply]
That's right.

**Reply by gquerret** [reply]
Starting from 1.3.11, it's possible to change the build mode without restarting the language server. In your case, you could switch to "Modified files" mode, modify the `.i` (no dependencies will be recompiled automatically due to the build mode), then save the `.p` (which will be recompiled due to the build mode).
